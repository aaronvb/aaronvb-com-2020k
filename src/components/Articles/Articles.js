import React from 'react';
import styles from './Articles.module.scss';

type Props = {
  children: React.Node
};

const Articles = ({ children }: Props) => (
  <div className={styles['articles']}>
    <div className={styles['articles__inner']}>
      <h4 href="#articles" id="articles">Articles</h4>
      <div className={styles['articles__body']}>
        {children}
      </div>
    </div>
  </div>
);

export default Articles;