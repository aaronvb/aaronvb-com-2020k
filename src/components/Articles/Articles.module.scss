@import '../../assets/scss/variables';
@import '../../assets/scss/mixins';

.articles {
    max-width: $layout-post-width;
    margin-left: auto;
    margin-right: auto;
    &__inner {
        padding: 0px 20px;
    }
    h4 {
        margin: 0;
        @include padding(1, 0, 1, 0);
    }
    @include margin-bottom(1);
}

@include breakpoint-sm {
    .articles {
        &__inner {
            @include padding(0, 0, 0, 0);
            border-top: 1px $color-gray-border solid;
        }
    }
}

@include breakpoint-md {
    .articles {
        &__inner {
            @include padding(0, 0, 0,);
            border-top: 1px $color-gray-border solid;
        }
    }
}
