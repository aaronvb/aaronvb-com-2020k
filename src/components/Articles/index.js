// @flow strict
export { default } from './Articles';
