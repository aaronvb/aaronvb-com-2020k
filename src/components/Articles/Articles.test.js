// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import Articles from './Articles';

describe('Articles', () => {
  const props = {
    children: 'test',
    title: 'test',
  };

  it('renders correctly', () => {
    const tree = renderer.create(<Articles {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
