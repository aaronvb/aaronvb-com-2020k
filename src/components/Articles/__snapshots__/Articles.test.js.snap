// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles renders correctly 1`] = `
<div
  className="articles"
>
  <div
    className="articles__inner"
  >
    <h4
      href="#articles"
      id="articles"
    >
      Articles
    </h4>
    <div
      className="articles__body"
    >
      test
    </div>
  </div>
</div>
`;
